app/keepers.go

```diff
@@ -506,10 +506,12 @@ func (app *NibiruApp) InitKeepers(
 
 	app.TokenFactoryKeeper = tokenfactorykeeper.NewKeeper(
 		keys[tokenfactorytypes.StoreKey],
+		keys[banktypes.StoreKey],
```

Comment:
If we don't need to ever delete denom metadata completely, then we also don't have to give the tokenfactory keeper access to the bank storeKey, thus preserving OCap semantics.

Reply:
Updated to simplify and remove here 39a9cc5a641aea86913b1ef93691631433e9d132